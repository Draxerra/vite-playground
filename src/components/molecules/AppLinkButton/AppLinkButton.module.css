@import "/src/css/abstracts/media-queries.css";

.app-link-button {
  text-decoration: none;
  color: inherit;
  box-shadow: none;
  border-radius: 0;

  &:not(.app-link-button--active) {
    color: var(--clr-primary-text);
    background-color: var(--clr-primary-surface2);
  }

  &:first-child {
    border-start-start-radius: 8px;
    border-end-start-radius: 8px;
  }

  &:last-child {
    border-start-end-radius: 8px;
    border-end-end-radius: 8px;
  }

  &:focus-visible {
    outline-offset: 0;
  }

  &:hover {
    --clr-primary-lightness: 50%;
    background-color: var(--clr-primary);
  }
}
