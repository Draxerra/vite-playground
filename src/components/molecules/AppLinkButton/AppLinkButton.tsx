import React, { ComponentProps, FC } from "react";
import { NavLink } from "react-router-dom";
import classNames from "classnames";

import AppButton from "~/components/atoms/AppButton";

import styles from "./AppLinkButton.module.css";

type AppLinkButtonProps = ComponentProps<NavLink> &
  ComponentProps<typeof AppButton>;
type AppAnchorButtonProps = AppLinkButtonProps & { navigate: () => void };

const AppAnchorButton: FC<AppAnchorButtonProps> = ({
  navigate,
  tag = "a",
  ...props
}) => (
  <AppButton
    tag={tag}
    onClick={(e) => {
      e.preventDefault();
      navigate();
    }}
    {...props}
  >
    {props.children}
  </AppButton>
);

const AppLinkButton: FC<AppLinkButtonProps> = ({
  activeClassName,
  className,
  ...props
}) => (
  <NavLink
    activeClassName={classNames(activeClassName, styles.appLinkButtonActive)}
    className={classNames(className, styles.appLinkButton)}
    component={AppAnchorButton}
    {...props}
  />
);

export default AppLinkButton;
