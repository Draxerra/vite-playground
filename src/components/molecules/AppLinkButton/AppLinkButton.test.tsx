import React from "react";
import AppLinkButton from "./AppLinkButton";
import { BrowserRouter as Router } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("App Link Button", () => {
  it("should match snapshot", () => {
    const linkButton = render(<AppLinkButton to="/">Home</AppLinkButton>, {
      wrapper: Router,
    }).asFragment();
    expect(linkButton).toMatchSnapshot();
  });

  it("should navigate on click", () => {
    render(<AppLinkButton to="/contact">Contact</AppLinkButton>, {
      wrapper: Router,
    });
    const linkButton = screen.getByText("Contact");
    expect(linkButton).not.toHaveClass("app-link-button--active");
    userEvent.click(linkButton);
    expect(linkButton).toHaveClass("app-link-button--active");
  });
});
