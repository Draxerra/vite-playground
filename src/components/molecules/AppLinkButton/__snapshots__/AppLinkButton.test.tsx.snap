// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Link Button should match snapshot 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="appLinkButton appLinkButtonActive appButton"
    href="/"
  >
    Home
  </a>
</DocumentFragment>
`;
